lib/iris/cube.py

```diff
@@ -28,6 +28,7 @@
 import numpy as np
 import numpy.ma as ma
 from packaging.version import Version
+from toolz import isiterable
```

Comment:
I have to admit, I was not aware of the `toolz` package.
I don't see its use any where else in the codebase. I see the use of `collections.abc.Iterable` in a `isinstance` test, which is the how I would have (naïvely) done it. Not saying you should do that, just pointing it out for you to consider (maybe there are some functionality differences?)